libpod/container_inspect.go

```diff
@@ -497,3 +880,458 @@ func (c *Container) generateInspectContainerConfig(spec *spec.Spec) (*InspectCon
+	binds := []string{}
+	for _, namedVol := range namedVolumes {
+		if len(namedVol.Options) > 0 {
+			binds = append(binds, fmt.Sprintf("%s:%s:%s", namedVol.Name, namedVol.Dest, strings.Join(namedVol.Options, ",")))
+		} else {
+			binds = append(binds, fmt.Sprintf("%s:%s", namedVol.Name, namedVol.Dest))
+		}
+	}
+	for _, mount := range mounts {
+		if mount.Type == "tmpfs" {
+			continue
+		}
+		// TODO - maybe we should parse for empty source/destination
+		// here. Would be confusing if we print just a bare colon.
+		if len(mount.Options) > 0 {
+			binds = append(binds, fmt.Sprintf("%s:%s:%s", mount.Source, mount.Destination, strings.Join(mount.Options, ",")))
+		} else {
+			binds = append(binds, fmt.Sprintf("%s:%s", mount.Source, mount.Destination))
+		}
+	}
+	hostConfig.Binds = binds
+
+	// Network mode parsing.
+	networkMode := ""
+	if c.config.CreateNetNS {
+		networkMode = "default"
+	} else if c.config.NetNsCtr != "" {
+		networkMode = fmt.Sprintf("container:%s", c.config.NetNsCtr)
+	} else {
+		// Find the spec's network namespace.
+		// If there is none, it's host networking.
+		// If there is one and it has a path, it's "ns:".
+		foundNetNS := false
+		for _, ns := range ctrSpec.Linux.Namespaces {
+			if ns.Type == spec.NetworkNamespace {
+				foundNetNS = true
+				if ns.Path != "" {
+					networkMode = fmt.Sprintf("ns:%s", ns.Path)
+				} else {
+					networkMode = "none"
+				}
+				break
+			}
+		}
+		if !foundNetNS {
+			networkMode = "host"
+		}
+	}
+	hostConfig.NetworkMode = networkMode
+
+	// Port bindings.
+	// Only populate if we're using CNI to configure the network.
+	portBindings := make(map[string][]InspectHostPort)
+	if c.config.CreateNetNS {
+		for _, port := range c.config.PortMappings {
+			key := fmt.Sprintf("%d/%s", port.ContainerPort, port.Protocol)
+			if _, ok := portBindings[key]; !ok {
+				portBindings[key] = []InspectHostPort{}
+			}
+			hostPorts := portBindings[key]
+			hostPorts = append(hostPorts, InspectHostPort{
+				HostIP:   port.HostIP,
+				HostPort: fmt.Sprintf("%d", port.HostPort),
+			})
+			portBindings[key] = hostPorts
+		}
+	}
+	hostConfig.PortBindings = portBindings
+
+	// Cap add and cap drop.
+	// We need a default set of capabilities to compare against.
+	// The OCI generate package has one, and is commonly used, so we'll
+	// use it.
+	// Problem: there are 5 sets of capabilities.
+	// Use the bounding set for this computation, it's the most encompassing
+	// (but still not perfect).
+	// TODO do not include when privileged is set (or compare against full
+	// caps?)
+	capAdd := []string{}
+	capDrop := []string{}
+	// No point in continuing if we got a spec without a Process block...
+	if ctrSpec.Process != nil {
+		// Max an O(1) lookup table for default bounding caps.
+		boundingCaps := make(map[string]bool)
+		g, err := generate.New("linux")
+		if err != nil {
+			return nil, err
+		}
+		for _, cap := range g.Config.Process.Capabilities.Bounding {
```

Comment:
@rhatdan PTAL at this section. I'm thinking Bounding is the safest set of capabilities to compute against, but I could be wrong.

Reply:
Best we got.